core/src/test/java/org/elasticsearch/fieldstats/FieldStatsTests.java

```diff
@@ -519,49 +600,84 @@ public void testMetaFieldsNotIndexed() {
     }
 
     public void testSerialization() throws IOException {
+        Version version = randomBoolean() ? Version.CURRENT : Version.V_5_1_0_UNRELEASED;
         for (int i = 0; i < 20; i++) {
-            assertSerialization(randomFieldStats());
+            assertSerialization(randomFieldStats(version.onOrAfter(Version.V_5_2_0_UNRELEASED)),
+                version);
         }
     }
 
     /**
      * creates a random field stats which does not guarantee that {@link FieldStats#maxValue} is greater than {@link FieldStats#minValue}
      **/
-    private FieldStats randomFieldStats() throws UnknownHostException {
+    private FieldStats randomFieldStats(boolean withNullMinMax) throws UnknownHostException {
         int type = randomInt(5);
```

Comment:
I know it isn't a part of this change but I'm not a fan of the `int type = randomInt(5); switch (type) {...}` idiom because I think it is easy to make a mistake that excludes a branch. I'd prefer `randomFrom(() -> {...}, () -> {...}).get()` because it is more obvious that you haven't forgot a branch. Is that crazy?